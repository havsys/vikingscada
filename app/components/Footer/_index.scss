.main-footer {
  background: #fff;
  padding: 0 21px;
  position: fixed;
  bottom: 0;
  width: 100%;
  .m-footer__copyright {
    font-family: Poppins;
    color:  #a9a9aa;
    font-size: 12px;
    font-weight: 400;
    line-height: 40px;

    a {
      &:hover {
        &:after {
          display: none;
        }
      }
      img {
        position: relative;
        margin-left: 6px;
        top: -4px;
      }
    }
  }
}

@media (max-width: 1024px) {
  .m-stack.m-stack--flex-tablet-and-mobile {
    display: table;
  }
  .m-stack.m-stack--flex-tablet-and-mobile.m-stack--ver>.m-stack__item.m-stack__item--last {
    display: table-cell;
    text-align: left;
  }
  .m-stack.m-stack--flex-tablet-and-mobile.m-stack--ver>.m-stack__item.m-stack__item--first {
    display: table-cell;
    text-align: right;
  }
}

@media (max-width: 768px) {
  .m-portlet {
    box-shadow: none !important;
  }
  .m-stack.m-stack--flex-tablet-and-mobile {
    display: flex;
  }
  .m-stack.m-stack--flex-tablet-and-mobile.m-stack--ver > .m-stack__item.m-stack__item--last {
    display: table-cell;
    text-align: left;
  }
  .m-stack.m-stack--flex-tablet-and-mobile.m-stack--ver > .m-stack__item.m-stack__item--first {
    display: table-cell;
    text-align: left;
  }
  .m-datatable.m-datatable--default>.m-datatable__pager>.m-datatable__pager-info {
    float: none; 
    display: block;
    margin: 10px 0 0 0;

    .m-datatable__pager-detail {
      .pager-detail__label {
        display: none;
      }
    }
  }
}