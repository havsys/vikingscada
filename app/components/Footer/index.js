import React from 'react';

/**
 * Footer component
 */
export class Footer extends React.Component {
  render() {
    return (
      <footer className="m-grid__item main-footer">
        <div className="m-container m-container--fluid m-container--full-height m-page__container">
          <div className="m-stack m-stack--flex-tablet-and-mobile m-stack--ver m-stack--desktop">
            <div className="m-stack__item m-stack__item--left m-stack__item--middle m-stack__item--last">
              <span className="m-footer__copyright">{`Â© ${new Date().getFullYear()} Viking SCADA `}</span>
            </div>
            <div className="m-stack__item m-stack__item--right m-stack__item--middle m-stack__item--first">
              <span className="m-footer__copyright">
                <span className="m-footer__copyright">
                  Powered by
                  <a href="https://indeema.com/" target="blank" className="m-link">
                    <img alt="" src="/img/indeema-logo.png" />
                  </a>
                </span>
              </span>
            </div>
          </div>
        </div>
      </footer>
    );
  }
}

export default Footer;
