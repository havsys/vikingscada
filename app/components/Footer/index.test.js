import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Footer from './index';

import Adapter from 'enzyme-adapter-react-16';

import renderer from 'react-test-renderer';


describe('>>>Footer --- Renders Footer message to user',()=>{
  let footer;
  
  const copyright = 'Â© 2018 Viking SCADA';

  Enzyme.configure({ adapter: new Adapter() })

  beforeEach(()=>{
      footer = shallow(
        <Footer/>
    ) 
  })

  it('+++ render the DUMB component', () => {
     expect(footer.length).toEqual(1)
  });

  it('+++capturing Snapshot of Footer', () => {

    const footer = renderer.create(
      <Footer/>
    );
    let tree = footer.toJSON();
    expect(tree).toMatchSnapshot();
  });
 
});