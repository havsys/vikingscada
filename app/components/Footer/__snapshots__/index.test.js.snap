// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Footer --- Renders Footer message to user +++capturing Snapshot of Footer 1`] = `
<footer
  className="m-grid__item main-footer"
>
  <div
    className="m-container m-container--fluid m-container--full-height m-page__container"
  >
    <div
      className="m-stack m-stack--flex-tablet-and-mobile m-stack--ver m-stack--desktop"
    >
      <div
        className="m-stack__item m-stack__item--left m-stack__item--middle m-stack__item--last"
      >
        <span
          className="m-footer__copyright"
        >
          Â© 2018 Viking SCADA 
        </span>
      </div>
      <div
        className="m-stack__item m-stack__item--right m-stack__item--middle m-stack__item--first"
      >
        <span
          className="m-footer__copyright"
        >
          <span
            className="m-footer__copyright"
          >
            Powered by
            <a
              className="m-link"
              href="https://indeema.com/"
              target="blank"
            >
              <img
                alt=""
                src="../img/indeema-logo.png"
              />
            </a>
          </span>
        </span>
      </div>
    </div>
  </div>
</footer>
`;
